src/MySqlConnector/MySqlDataReader.cs

```diff
@@ -474,8 +474,13 @@ internal static async Task<MySqlDataReader> CreateAsync(CommandListPosition comm
 				await dataReader.NextResultAsync(ioBehavior, cancellationToken).ConfigureAwait(false);
 			}
 		}
-		catch (Exception)
+		catch (Exception ex)
 		{
+			if (activity is { IsAllDataRequested: true })
+			{
+				activity.SetException(ex);
+				activity.Stop();
+			}
 			dataReader.Dispose();
```

Comment:
`MySqlDataReader.DisposeAsync` will still Stop the activity and set it to successful. Seems like that logic ought to change to if `if (Activity.Status == ActivityStatusCode.Unset)` to avoid overwriting the status if it's already an error.